### PR TITLE
refactor: single dataset mode

### PR DESCRIPTION
**Pull request by Bento007**
- #181
#### Reviewers
**Functional:** 

**Readability:** 

---

## Changes
- Remove single dataset mode parameters from config and update comments
- Remove single dataset mode if conditions
- multi-dataset mode is required in the config. An error is raised if it's not configured.
- Removing unused test functions
- Remove single dataset mode tests

### CODE REVIEW

File: server/dataset/dataset_metadata.py [hunk 1]
```diff
@@ -34,11 +34,8 @@ def infer_dataset_s3_uri(server_config: ServerConfig, dataset_root: str, dataset
     Use the dataset_root, dataset_id, and the server config to infer the physical S3 URI for an Explorer dataset
     artifact
     """
-    # TODO @mdunitz remove after fork, update config to remove single_dataset option, the multiroot lookup will need to
-    #  remain while we support covid 19 cell atlas.
+    # TODO @mdunitz remove after fork, the multiroot lookup will need to remain while we support covid 19 cell atlas.
```
Comment (metakuni) [inline server/dataset/dataset_metadata.py:37]: I guess this TODO (`@mdunitz`) will never get done 😆

File: server/common/config/server_config.py [hunk 2]
```diff
@@ -143,7 +135,7 @@ def handle_data_locator(self):
                 for path in paths:
                     if path.startswith("s3://"):
                         break
-            if path.startswith("s3://"):
+            if isinstance(path, str) and path.startswith("s3://"):
```
Comment (atolopko-czi) [inline server/common/config/server_config.py:138]: bug fix?

File: server/tests/unit/__init__.py [hunk 3]
```diff
@@ -3,43 +3,23 @@
...
-
-        return wraps
-
-    return decorator
-
-
-def app_config(data_locator, extra_server_config={}, extra_dataset_config={}):
+def app_config(extra_server_config=None, extra_dataset_config=None):
+    extra_server_config = extra_server_config if extra_server_config else {}
+    extra_dataset_config = extra_dataset_config if extra_dataset_config else {}
```
Comment (atolopko-czi) [inline server/tests/unit/__init__.py:16]:
```suggestion
    extra_dataset_config = extra_dataset_config or {}
```

Comment (atolopko-czi) [reply server/tests/unit/__init__.py:16]: curious why these new lines are now needed?

**Review by atolopko-czi — CHANGES_REQUESTED** [review]
All of the removals look reasonable. (Admittedly, it's hard to see the big picture w/o a deeper code dive.) 

Before approving, what do you think of removing the term "multi_dataset" everywhere, now that we no longer need to differentiate between the single/multi modes?

File: server/common/config/server_config.py (hunk 2 above)
Comment (Bento007) [reply server/common/config/server_config.py:138]: yes, in some tests the path is None.

File: server/dataset/dataset_metadata.py (hunk 1 above)
Comment (Bento007) [inline server/dataset/dataset_metadata.py:37]:
```suggestion
```

File: server/tests/unit/__init__.py (hunk 3 above)
Comment (Bento007) [reply server/tests/unit/__init__.py:16]: those parameters were default set to `{}` which is mutable and could cause strange behavior if manipulated. I changed the defaults to `None` which is immutable, and set them to `{}` if `None`

**Comment by Bento007** [comment]
I think removing `multi_dataset` would be a mistake since the name is used to parse the config, and there are setting nested under the mult_dataset setting. Changing the name is an option, but I can't think of a better name. When we clean up the config, a better name might become easier to decide on or obsolete.

File: server/dataset/dataset_metadata.py [hunk 4]
```diff
@@ -34,11 +34,7 @@ def infer_dataset_s3_uri(server_config: ServerConfig, dataset_root: str, dataset
     Use the dataset_root, dataset_id, and the server config to infer the physical S3 URI for an Explorer dataset
     artifact
     """
-    # TODO @mdunitz remove after fork, update config to remove single_dataset option, the multiroot lookup will need to
-    #  remain while we support covid 19 cell atlas.
     #   See ticket https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/corpora-data-portal/1281 # noqa
```
Comment (atolopko-czi) [inline server/dataset/dataset_metadata.py:37]: Not sure that this issue reference is useful to understanding. Remove? If should keep, at least update to the correct ticket (it moved repos):
```
    #   See ticket https://github.com/chanzuckerberg/single-cell-explorer/issues/119 # noqa
```

**Review by atolopko-czi — APPROVED** [review]

**Pull request merged**